### PR TITLE
UI Optimizations

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...
- Combine display icon button templates in display container template
- Create DOM elements with `document.create` and `className =` instead of parsing small templates for simple tooltips and custom buttons
- Improve how controlbar handles updates to custom buttons. It was parsing and replacing the related playlist button on each state change from related. Now it only replaces buttons when a button with a new id or button class is introduced
- Combine parsing of multiple SVG elements into one call and extract icons with query selectors
- Don't add styles to the DOM when there are no color customization options set

### Why is this Pull Request needed?
`utils.createElement` (and any code that results in "Parse HTML" operations) is costly memory and processor wise. Reducing the number of calls by combining templates or creating 1-2 elements at a time with `document.createElement('div')` is better than using templates for simple DOM structures

There is only one display container and set of icons, so it doesn't make sense to parse the icon buttons outside of parsing the container - It's one component. Parsing it at once reduces the number of nodes and memory required to enable controls.

Simple tooltips and custom buttons are composed of two elements (or one container and an SVG document). The DOM elements for these components are created much faster and with less memory overhead by avoiding html parsing.

JW8-436

### CODE REVIEW

File: src/js/view/controls/controlbar.js [hunk 1]
```diff
@@ -499,10 +519,17 @@ export default class Controlbar {
 
     updateButtons(model, newButtons = [], oldButtons = []) {
         const buttonContainer = this.elements.buttonContainer;
-        this.removeButtons(buttonContainer, oldButtons);
 
-        for (let i = newButtons.length - 1; i >= 0; i--) {
-            let buttonProps = newButtons[i];
+        const removedButtons = oldButtons.filter(oldButton =>
+            !newButtons.some(newButton => newButton.id + newButton.btnClass === oldButton.id + oldButton.btnClass));
```
Comment (karimJWP) [inline src/js/view/controls/controlbar.js]: `includes` definitely wouldn't work here ?

Comment (robwalch) [reply src/js/view/controls/controlbar.js]: Includes is ES6 and I'm not sure how or if Babel would transform it https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Specifications

Also the object instances are likely not the same, so comparing properties (`id` and `btnClass`) with `some` gets it done.

Comment (robwalch) [reply src/js/view/controls/controlbar.js]:
```js
[3].includes(3)
// true

[{a: 3}].includes({a: 3})
// false

[{a: 3}].some(o => o.a === {a: 3}.a)
// true
```

Comment (karimJWP) [reply src/js/view/controls/controlbar.js]: 👍

**Comment by robwalch** [comment]
These changes address some issues in #2054

**Pull request merged**